### PR TITLE
Consul support

### PR DESCRIPTION
**Pull request by ruslansennov**
@cescoffier please review

### CODE REVIEW

File: vertx-config-consul/src/main/java/io/vertx/config/consul/package-info.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
+ * === Consul Configuration Store
+ *
+ * The Consul Configuration Store extends the Vert.x Configuration Retriever and provides the
+ * way to retrieve configuration from a Consul.
```
Comment (cescoffier) [inline vertx-config-consul/src/main/java/io/vertx/config/consul/package-info.java]: Maybe add a link to https://www.consul.io/.

File: vertx-config-consul/src/test/java/io/vertx/config/consul/ConsulConfigStoreTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  private void createRetriever() {
+    retriever = ConfigRetriever.create(vertx,
+      new ConfigRetrieverOptions().addStore(
+        new ConfigStoreOptions()
+          .setType("consul")
+          .setConfig(new JsonObject()
+            .put("port", consulProcess.getHttpPort())
+            .put("prefix", "foo"))));
+  }
```
Comment (cescoffier) [inline vertx-config-consul/src/test/java/io/vertx/config/consul/ConsulConfigStoreTest.java]: Could you add a test with a config listener and a config stream to be sure that changes to the configuration stored in consul are caught.

File: vertx-config-consul/src/main/asciidoc/ruby/consul-store.adoc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    'prefix' => "foo"
+  }
+}
+
+retriever = VertxConfig::ConfigRetriever.create(vertx, {
+  'stores' => [
+    store
+  ]
+})
+
```
Comment (cescoffier) [inline vertx-config-consul/src/main/asciidoc/ruby/consul-store.adoc]: Can you add a line about `prefix` and `delimiter`.

**Comment by ruslansennov** [comment]
@cescoffier thanks for comments, done

**Comment by cescoffier** [comment]
Thanks ! That's awesome!

**Comment by vietj** [comment]
nice!

**Pull request merged**